File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int left = 0; 
        int right = s.size() - 1; 
        string check = "AEIOUaeiou"; 

        while (left <= right){
            
            if(check.find(s[right]) != string::npos && check.find(s[left]) != string::npos){
                int temp = s[right]; 
                s[right] = s[left]; 
                s[left] = temp;  
                right--; 
                left++; 
            }else if(check.find(s[left]) == string::npos){
                left++; 
            }else if (check.find(s[right]) == string::npos){
                right--; 
            }
        }
        return s; 
        
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = "aeiouAEIOU"
        arr = list(s)
        left, right = 0, len(arr)-1

        while left < right: 
            if arr[right] in vowels and s[left] in vowels: 
                arr[right], arr[left] = arr[left], arr[right]
                right -= 1
                left += 1

            if arr[right] not in vowels: 
                right -=1
            if arr[left] not in vowels: 
                left += 1
            
            
        
        return "".join(arr)